Makefile

```
@@ -0,0 +1,252 @@
+
+RESOURCES := $(wildcard lambda/templates/*)
+TERRAFORM := $(wildcard terraform/*)
+
+# Output directory
+DIR := dist
+EMPTY := $(DIR)/empty
+# Temporary artifacts
+DIST_SOURCES := $(SOURCES:lambda/%=$(DIR)/code/%)
+DIST_RESOURCES := $(RESOURCES:lambda/%=$(DIR)/code/%)
+DIST_TERRAFORM := $(TERRAFORM:terraform/%=$(DIR)/terraform/%)
+
+BUCKET_MAP_OBJECT_KEY := DEFAULT
+
+DATE := $(shell date -u "+%b %d %Y, %T %Z")
+DATE_SHORT := $(shell date -u "+%Y%m%dT%H%M%S")
+BUILD_ID := $(shell git rev-parse --short HEAD 2>/dev/null || echo "SNAPSHOT")
+
+DOCKER := docker
+# On Linux we need to do a bit of userid finagling so that the output files
+# end up being owned by us and not by root. On Mac this works out of the box.
+DOCKER_USER_ARG := --user "$(shell id -u):$(shell id -g)"
+DOCKER_COMMAND := $(DOCKER) run --rm $(DOCKER_USER_ARG) -v "$$PWD":/var/task lambci/lambda:build-python3.8
+
+#####################
+# Deployment Config #
+#####################
+
+# A tag to distinguish between artifacts of the same name. The tag should be
+# different for each build.
+S3_ARTIFACT_TAG = $(DATE_SHORT)
+
+
+# Include custom configuration
+Makefile.config:
+	@echo
+	@echo "It looks like you are building TEA for the first time."
+	@echo "Please review the configuration in '$@' and run Make again."
+	@echo
+	@test -e $@ || cp Makefile.config.example $@
+	@exit 1
+
+include Makefile.config
+
+
+.DEFAULT_GOAL := all
+.PHONY: all
+all: build ;
+
+##############################
+# Local building/development #
+##############################
+
+# Build everything
+.PHONY: build
+build: \
+	$(DIR)/thin-egress-app-code.zip \
+	$(DIR)/thin-egress-app-dependencies.zip \
+	$(DIR)/thin-egress-app.yaml \
+	$(DIR)/thin-egress-app-terraform.zip
+
+# Build individual components
+.PHONY: dependencies
+dependencies: $(DIR)/thin-egress-app-dependencies.zip
+	@echo "Built dependency layer for version ${BUILD_ID}"
+
+.PHONY: code
+code: $(DIR)/thin-egress-app-code.zip
+	@echo "Built code for version ${BUILD_ID}"
+
+.PHONY: yaml
+yaml: $(DIR)/thin-egress-app.yaml
+	@echo "Built CloudFormation template for version ${BUILD_ID}"
+
+.PHONY: terraform
+terraform: $(DIR)/thin-egress-app-terraform.zip
+	@echo "Build Terraform zip file for version ${BUILD_ID}"
+
+.PHONY: clean
+clean:
+	rm -rf $(DIR)
+
+$(DIR)/thin-egress-app-dependencies.zip: requirements.txt
+	rm -rf $(DIR)/python
+	@mkdir -p $(DIR)/python
+	$(DOCKER_COMMAND) build/dependency_builder.sh "$(DIR)/thin-egress-app-dependencies.zip" "$(DIR)"
+
+.SECONDARY: $(DIST_RESOURCES)
+$(DIST_RESOURCES): $(DIR)/code/%: lambda/%
+	@mkdir -p $(@D)
+	cp $< $@
+
+.SECONDARY: $(DIST_SOURCES)
+$(DIST_SOURCES): $(DIR)/code/%: lambda/%
+	@mkdir -p $(@D)
+	cp $< $@
+	python3 scripts/sed.py -i $@ "<BUILD_ID>" "${BUILD_ID}"
+
+$(DIR)/thin-egress-app-code.zip: $(DIST_SOURCES) $(DIST_RESOURCES)
+	@mkdir -p $(DIR)/code
+	cd $(DIR)/code && zip -r ../thin-egress-app-code.zip .
+
+$(DIR)/bucket-map.yaml: | config/bucket-map-template.yaml
+	cp -i $| $@
```

This probably makes mores sense TBH
```suggestion
$(DIR)/bucket-map.yaml:
	cp -i config/bucket-map-template.yaml $@
```